OWASP/SecureTea-Project
# Current users notifs

## Status
**READY**

## Description
It introduces the feature to report the users logged into the current server. What it reports is the users logged in the system. Whenever a user is added or logs out, the user is notified for the same. To make it optimal a table called USERS is created which contains the data of users currently logged in. If a user logs out, the corresponding entry is deleted.

Threading has also been done. Since a sqlite variable can only be accesed in the same thread, hence the code is optimised accordingly. Hence there is no requirement of introducing locks.

Sample Update:
![Sample-Update](https://i.ibb.co/HtGWvH9/UPDATE.png)
## Related PRs
List related PRs against other branches:

branch | PR
------ | ------
other_pr_production | [link]()
other_pr_master | [link]()


## Todos


## Deploy Notes
None
## Steps to Test or Reproduce
Run the code choosing a notification medium of your choice. Get notifications using users added. SSH to the same server using ssh-server and see the user getting updated.

```sh
git pull --prune
git checkout <feature-branch>
```

1. 

## Impacted Areas in Application
securetea/core.py
securetea/users.py
*
